serverless/src/triggers/auth/createAuthChallenge/index.js

```diff
@@ -1,15 +1,30 @@
 const crypto = require('crypto-secure-random-digit');
 const { apiKey, apiSecret } = require('../../../../mailjetConfig');
 const mailjet = require('node-mailjet').connect(apiKey, apiSecret);
+const { getUser } = require('../../../shared/users');
+
+const THREE_MINUTES = 3 * 60 * 1000;
 
 const handler = async event => {
   let secretLoginCode;
   if (!event.request.session || !event.request.session.length) {
     // This is a new auth session
-    // Generate a new secret login code and mail it to the user
-    secretLoginCode = crypto.randomDigits(6).join('');
 
-    await sendEmail(event.request.userAttributes, secretLoginCode);
+    // Get user to check if a token was added recently (sub is userId)
+    const result = await getUser(event.request.userAttributes.sub);
+
+    // Only set the token as loginCode if it is not older than 10 minutes
```

Comment:
Here it is commented: "not older than 10 minutes" but below the check is performed against "THREE_MINUTES"

Reply:
Oh thanks forgot to change that I guess. This will be part of the next pull request anyway :)